walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,192 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // The Collection that stores the available times for the requested meeting.
+    Collection<TimeRange> availableTimes = new ArrayList<TimeRange>();
+
+    // Check to see if the duration is longer than a day.
+    if (request.getDuration() == TimeRange.WHOLE_DAY.duration() + 1) {
+      return availableTimes;
+    }
+    
+    List<TimeRange> unavailableTimes = getSortedUnavailableTimes(events, request);
+    availableTimes.addAll(computeAvailableTimes(unavailableTimes, request));
```

Comment:
why not just return computeAvailableTimes?

Reply:
Gotcha, I'll change that